uwp/uwp_main.cpp

```diff
@@ -638,6 +681,57 @@ void App::OnPackageInstalling(PackageCatalog^ sender, PackageInstallingEventArgs
 	}
 }
 
+void App::ParseProtocolArgs(Windows::ApplicationModel::Activation::IActivatedEventArgs^ args, int *argc, std::vector<char*> *argv, std::vector<std::string> *argvTmp)
+{
+	argvTmp->clear();
+	argv->clear();
+
+	if (args->Kind == ActivationKind::Protocol)
+	{
+		ProtocolActivatedEventArgs^ protocolArgs = dynamic_cast<Windows::ApplicationModel::Activation::ProtocolActivatedEventArgs^>(args);
+		Windows::Foundation::WwwFormUrlDecoder^ query = protocolArgs->Uri->QueryParsed;
+
+		//if RetroArch UWP app is started using protocol with argument "launchOnExit", this gives an option to launch another app on RA exit,
+		//making it easy to integrate RA with other UWP frontends
+		try
+		{
+			m_launchOnExit = query->GetFirstValueByName("launchOnExit");
+		}
+		catch (Platform::InvalidArgumentException^ e)
+		{
+			//nothing to do if named parameter doesn't exist
+		}
+
+		try
+		{
+			//protocol activation is in format 0=argValue0&1=argValue1...
+			//where argument name is a number from 0 to 9
+			for (int i = 0; i < 10; i++)
```

Comment:
I still feel like this should be done via a html encoded string or a list of args stored as an actual list in the url like
["arg","value"] rather than each argument being a separate query param

Reply:
I understand. I think It's doable, If we go with the html encoded strings, here's a few examples how it could work:
1. just one parameter containing all in one line, something like.
not encoded> retroarch:?retroarch -L cores\stella_libretro.dll "c:\mypath\My Files\game.rom" launchOnExit=LaunchApp:
encoded> retroarch:?retroarch%20-L%20cores%5Cstella_libretro.dll%20%22c%3A%5Cmypath%5CMy%20Files%5Cgame.rom%22%20launchOnExit%3DLaunchApp%3A

2. one parameter for retroarch args and one for launchOnExit parameter since it's not a standard argument supported by all platforms but just UWP, so something like:
not encoded> retroarch:?cmd=retroarch -L cores\stella_libretro.dll "c:\mypath\My Files\game.rom"&launchOnExit=LaunchApp:
encoded> retroarch:?cmd=retroarch%20-L%20cores%5Cstella_libretro.dll%20%22c%3A%5Cmypath%5CMy%20Files%5Cgame.rom%22&launchOnExit=LaunchApp:

Let me know what is proffered.

Reply:
Probably something more like the second, so that additional new uwp specific parameters can be added in future

Reply:
Implemented 2nd option.